lib/extensions/virtual_display_name.rb

```diff
@@ -0,0 +1,15 @@
+module VirtualDisplayName
+
+  # define an attribute to contain display text, in order of preference:
+  # description, if and when available
+  # title/name in case description is not available or does not exist
+  # default to <record id> in cases where no other attribute has been provided by user
+  def virtual_display_name
+    return label                      if respond_to?("label")
+    return description                if respond_to?("description") && description.present?
+    return ext_management_system.name if respond_to?("ems_id")
```

Comment:
Should this be `if respond_to?(:ext_management_system)`?  The underlying column may not necessarily be `ems_id`.

Even so, this is wrong.  Many inventory objects are linked to EMSes, and this will cause their name to be the EMS...even worse *all* objects of that type will be the EMS name so they all look the same.  Example:

```
[7] pry(main)> EmsCluster.first.virtual_display_name
=> "dev-vc60.cloudforms.lab.eng.rdu2.redhat.com"
[8] pry(main)> EmsCluster.first.name
=> "test-vc60-cluster"
```

@dclarizio Can you comment here on the expectations?

Reply:
@Fryguy agree, @h-kataria and I are looking into the origin of that line that pulls the ems name.